exp/operator/controllers/genericprovider_controller.go

```diff
@@ -92,14 +104,175 @@ func (r *GenericProviderReconciler) Reconcile(ctx context.Context, req reconcile
 	return r.reconcile(ctx, typedProvider, typedProviderList)
 }
 
+func (r *GenericProviderReconciler) secretReader(ctx context.Context, genericProvider genericprovider.GenericProvider) (configclient.Reader, error) {
+	mr := configclient.NewMemoryReader()
+
+	// TODO: maybe set a shorter default, so we don't block when installing cert-manager.
+	//mr.Set("cert-manager-timeout", "120s")
+
+	if genericProvider.GetSpec().SecretName != nil {
```

Comment:
maybe we should add a log message that secret is missing?

Reply:
OK, will do.